src/main/java/no/finn/unleash/DefaultUnleash.java

```diff
@@ -56,6 +56,10 @@ public DefaultUnleash(UnleashConfig unleashConfig, Strategy... strategies) {
     }
 
     public DefaultUnleash(UnleashConfig unleashConfig, ToggleRepository toggleRepository, Strategy... strategies) {
+        this(unleashConfig, toggleRepository, unleashScheduledExecutor, strategies);
+    }
+
+    public DefaultUnleash(UnleashConfig unleashConfig, ToggleRepository toggleRepository, UnleashScheduledExecutor unleashScheduledExecutor, Strategy... strategies) {
```

Comment:
How is overloading the constructor with providing your own `UnleashScheduledExecutor` related to the issue this PR tries to solve?

Reply:
Its not directly related, but it did allow me to inject a mocked UnleashScheduledExecutor while testing this to prevent schedulation in the background. 

Also, I think it would be a nice change in general to be able to supply the same custom UnleashScheduledExecutor to DefaultUnleash as to FeatureToggleRepository in this manner:

```

UnleashScheduledExecutor unleashScheduledExecutor = new MyCustomUnleashScheduledExecutor();
        FeatureToggleRepository featureToggleRepository = new FeatureToggleRepository(
                unleashConfig,
                unleashScheduledExecutor,
                new HttpToggleFetcher(unleashConfig),
                new ToggleBackupHandlerFile(unleashConfig)
        );
        Unleash unleash = new DefaultUnleash(unleashConfig, featureToggleRepository, unleashScheduledExecutor);

```